libbeat/common/docker/watcher.go

```diff
@@ -316,7 +338,8 @@ func (w *watcher) listContainers(options types.ContainerListOptions) ([]*Contain
 		// If there are no network interfaces, assume that the container is on host network
 		// Inspect the container directly and use the hostname as the IP address in order
 		if len(ipaddresses) == 0 {
-			info, err := w.client.ContainerInspect(w.ctx, c.ID)
+			logp.Debug("docker", "Inspect container %s", c.ID)
+			info, err := w.client.ContainerInspect(ctx, c.ID)
```

Comment:
If we keep current behavior of only logging containers, that failed to be inspected, each inspect call must have separate `context.withTimeout` for its own call